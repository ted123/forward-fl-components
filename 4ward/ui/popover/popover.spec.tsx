import React from 'react';
import ReactDOM from 'react-dom';
import {
  render, cleanup,
} from '@testing-library/react';
import { Popover } from '.';

describe('Popover Component test', () => {
  const isClick = true;

  afterEach(cleanup);

  it('should render with the correct text', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Popover
        content="Lorem ipsum may be used as a placeholder before the final copy is available."
        position="bottom"
        color="#238749"
        show={isClick}
      >
        Hello world!
      </Popover>, div,
    );
    expect(div).toBeTruthy();
  });

  it('should show popover - bottom', () => {
    const tree = render(
      <Popover
        content="Lorem ipsum may be used as a placeholder before the final copy is available."
        position="bottom"
        color="#238749"
        show
      >
        Hello World!
      </Popover>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should show popover - top', () => {
    const tree = render(
      <Popover
        content="Lorem ipsum may be used as a placeholder before the final copy is available."
        position="top"
        color="#238749"
        show={isClick}
      >
        Hello World!
      </Popover>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should show popover - right', () => {
    const tree = render(
      <Popover
        content="Lorem ipsum may be used as a placeholder before the final copy is available."
        position="right"
        color="#238749"
        show={isClick}
      >
        Hello World!
      </Popover>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should show popover - left', () => {
    const tree = render(
      <Popover
        content="Lorem ipsum may be used as a placeholder before the final copy is available."
        position="left"
        color="#238749"
        show={isClick}
      >
        Hello World!
      </Popover>,
    );
    expect(tree).toMatchSnapshot();
  });
});
