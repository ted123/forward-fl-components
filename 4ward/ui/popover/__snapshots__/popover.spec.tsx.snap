// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover Component test should show popover - bottom 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL cgWYXr"
      >
        <div
          class="sc-hKwDye iGMhlI"
        >
          Hello World!
        </div>
        <div
          class="sc-gsDKAQ lbsyXb"
          color="#238749"
        >
          <div
            class="sc-dkPtRN bflDEl"
            color="#238749"
          >
            Lorem ipsum may be used as a placeholder before the final copy is available.
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL cgWYXr"
    >
      <div
        class="sc-hKwDye iGMhlI"
      >
        Hello World!
      </div>
      <div
        class="sc-gsDKAQ lbsyXb"
        color="#238749"
      >
        <div
          class="sc-dkPtRN bflDEl"
          color="#238749"
        >
          Lorem ipsum may be used as a placeholder before the final copy is available.
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Popover Component test should show popover - left 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL cgWYXr"
      >
        <div
          class="sc-hKwDye iGMhlI"
        >
          Hello World!
        </div>
        <div
          class="sc-gsDKAQ hQoPZI"
          color="#238749"
        >
          <div
            class="sc-dkPtRN bvcIbT"
            color="#238749"
          >
            Lorem ipsum may be used as a placeholder before the final copy is available.
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL cgWYXr"
    >
      <div
        class="sc-hKwDye iGMhlI"
      >
        Hello World!
      </div>
      <div
        class="sc-gsDKAQ hQoPZI"
        color="#238749"
      >
        <div
          class="sc-dkPtRN bvcIbT"
          color="#238749"
        >
          Lorem ipsum may be used as a placeholder before the final copy is available.
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Popover Component test should show popover - right 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL cgWYXr"
      >
        <div
          class="sc-hKwDye iGMhlI"
        >
          Hello World!
        </div>
        <div
          class="sc-gsDKAQ hVymLd"
          color="#238749"
        >
          <div
            class="sc-dkPtRN iKAXzw"
            color="#238749"
          >
            Lorem ipsum may be used as a placeholder before the final copy is available.
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL cgWYXr"
    >
      <div
        class="sc-hKwDye iGMhlI"
      >
        Hello World!
      </div>
      <div
        class="sc-gsDKAQ hVymLd"
        color="#238749"
      >
        <div
          class="sc-dkPtRN iKAXzw"
          color="#238749"
        >
          Lorem ipsum may be used as a placeholder before the final copy is available.
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Popover Component test should show popover - top 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL cgWYXr"
      >
        <div
          class="sc-hKwDye iGMhlI"
        >
          Hello World!
        </div>
        <div
          class="sc-gsDKAQ idrGcd"
          color="#238749"
        >
          <div
            class="sc-dkPtRN bCVybs"
            color="#238749"
          >
            Lorem ipsum may be used as a placeholder before the final copy is available.
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL cgWYXr"
    >
      <div
        class="sc-hKwDye iGMhlI"
      >
        Hello World!
      </div>
      <div
        class="sc-gsDKAQ idrGcd"
        color="#238749"
      >
        <div
          class="sc-dkPtRN bCVybs"
          color="#238749"
        >
          Lorem ipsum may be used as a placeholder before the final copy is available.
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
